Fix wrong .gitignore for Carthage

jechol opened the pull request:
`*/Carthage/Build/` pattern is not working.
Carthage/Build is correct pattern.

I confirmed that in https://github.com/ReactiveKit/Bond

wokalski commented:
Hm, I'm not sure. Notice that there's no `Cartfile` in the project and it should just ignore Carthage output for the subproject in `PerfTests`

jechol commented:
I mean when this project is included in other library with carthage,
current .gitignore doesn't ignore generated carthage related folder.

ReactiveKit/Bond, for example,
a soft link is created at `Bond/Carthage/Checkout/Diff.swift/Carthage/Build`,
which points to `Bond/Carthage/Build`.
But current .ignore can't ignore `Bond/Carthage/Checkout/Diff.swift/Carthage/Build`,
so Diff.swift submodule is marked as dirty, which is very annoying.

Please check out ReactiveKit/Bond, then run `carthage update`,
and you will notice what I mean.

Thanks!

wokalski commented:
I see the issue. I don't know why Carthage even creates the link. Any clues? I'd rather fix the issue than work around it.

jechol commented:
Carthage seems to make a folder for each included projects.

For example, when run `carthage update` on Bond,
`Carthage/Build` links are created in folders for both of its dependencies `ReactiveKit` and `Diff.swift`.

So ignoring seems the only way, and I don't think it's workaround.

Thanks.

wokalski commented:
Great, can we maintain both ignores then? I'd like git to ignore carthage build and checkout for the project in PerfTests, too.

jechol commented:
Added `**/` to rules, so it can ignore Carthage/Build and Checkouts recursively.
And also removed PerfTests specific rule.

Please check it again.
